### DISCUSSION TITLE
Is my google cloud setup any good?

### DISCUSSION THREAD

**Discussion by JulieVelghe** [craftcms/cms#15284] [General]
Hi all,
we've made a website for a client, where we now need to be able to upload files to a google cloud storage via a form.
My media system for audio files is setup like this:
![image](https://github.com/craftcms/cms/assets/46524074/f99f1090-b6c5-4e59-af25-d50642488410)
I am using variables from my env file:
```
GOOGLE_APPLICATION_CREDENTIALS="D:/Julie/Repos/Ritcs/erasmushogeschool-brussel-fb3422db4ecf.json"
GOOGLE_CLOUD_PROJECT_ID="erasmushogeschool-brussel"
GOOGLE_CLOUD_URL="https://storage.googleapis.com/ritcs-folio"
```
When I submit my form with an audio file, I get a filesystem error:
![image](https://github.com/craftcms/cms/assets/46524074/acdd7c31-414e-4916-b6c0-e26548773068)

Does anyone have a clue as to where I have to look for the error? Is it something in my settings, or something in the google cloud storage?

**Comment by brandonkelly** [comment]
Can you try uploading the same file but with a simplified filename with no dashes or spaces?

**Reply by JulieVelghe** [reply]
Ok, I tried that, but I got the same error.

**Reply by angrybrad** [reply]
Do the credentials in `erasmushogeschool-brussel-fb3422db4ecf.json` have the necessary permissions to write to the bucket?

**Reply by JulieVelghe** [reply]
hi @angrybrad , I think so. The google cloud storage was setup by another party, so I'll try to get that verified.

**Reply by JulieVelghe** [reply]
hi @angrybrad , we have acquired access to the google cloud storage. Could you steer me in the right direction, as to where I have to look for these permissions? Which one I have to set at what value and where?
Unfortunatly, the read-me for the craft plugin doesn't talk about the needed Google Cloud setup.

**Reply by JulieVelghe** [reply]
Hi @brandonkelly , we have acquired access to the google cloud storage. Our service account has the role 'Storage Admin', which should have enough rights to everything. However, this error is still being thrown. 
[Filesystem Error – craft_errors_FsException.pdf](https://github.com/user-attachments/files/16362513/Filesystem.Error.craft_errors_FsException.pdf)
The attached pdf shows the entire error stack. 
What else could be the issue here?

**Reply by angrybrad** [reply]
@JulieVelghe Since you [created an issue](https://github.com/craftcms/google-cloud/issues/40) on the Google Cloud plugin repo, I’ll close this discussion, and we can continue the conversation there.
